src/actions/segment/segment.ts

```diff
@@ -55,6 +58,34 @@ export class SegmentAction extends Hub.Action {
   requiredFields = [{ any_tag: this.allowedTags }]
   executeInOwnProcess = true
 
+  asJson(router: RouteBuilder, request?: any): any {
```

Comment:
This is a repetition of the entire `asJson` implementation from the superclass right? That seems extremely brittle. 

What if instead we did something like `supportedFormats` could optionally be a function like `this.supportedFormats = (request) => { ... }` that can modulate the supported formats based on the current request?